# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.h
# Commit: 336f58a54bd8
# Full Hash: 336f58a54bd8128c47fa783a508877096ba84918
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 2
# Description:
#   Bug 1654112 - Move AudioConduit from a Mutex to an RWLock. r=padenot
#   
#   AudioConduit will try to grab the lock on the audio thread, and if it fails
#   there is a risk of glitching. That is fine as long as the only other usage of
#   the lock is related to renegotiation where the audio stream is being recreated
# ==============================================================================

diff -r a802c8f458cb -r 336f58a54bd8 dom/media/webrtc/libwebrtcglue/AudioConduit.h
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Wed Aug 25 16:44:09 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Wed Aug 25 20:28:12 2021 +0200
@@ -7,6 +7,7 @@
 
 #include "mozilla/Attributes.h"
 #include "mozilla/ReentrantMonitor.h"
+#include "mozilla/RWLock.h"
 #include "mozilla/StateMirroring.h"
 #include "mozilla/TimeStamp.h"
 
@@ -198,14 +199,14 @@
   // Accessed only on the Call thread.
   webrtc::AudioReceiveStream::Config mRecvStreamConfig;
 
-  // Written only on the Call thread. Guarded by mMutex, except for reads on the
+  // Written only on the Call thread. Guarded by mLock, except for reads on the
   // Call thread.
   webrtc::AudioReceiveStream* mRecvStream;
 
   // Accessed only on the Call thread.
   webrtc::AudioSendStream::Config mSendStreamConfig;
 
-  // Written only on the Call thread. Guarded by mMutex, except for reads on the
+  // Written only on the Call thread. Guarded by mLock, except for reads on the
   // Call thread.
   webrtc::AudioSendStream* mSendStream;
 
@@ -216,18 +217,18 @@
   RtpPacketQueue mRtpPacketQueue;
 
   // If true => mSendStream started and not stopped
-  // Written only on the Call thread. Guarded by mMutex, except for reads on the
+  // Written only on the Call thread. Guarded by mLock, except for reads on the
   // Call thread.
   bool mSendStreamRunning;
   // If true => mRecvStream started and not stopped
-  // Written only on the Call thread. Guarded by mMutex, except for reads on the
+  // Written only on the Call thread. Guarded by mLock, except for reads on the
   // Call thread.
   bool mRecvStreamRunning;
 
   // Accessed only on the Call thread.
   bool mDtmfEnabled;
 
-  mutable Mutex mMutex;
+  mutable RWLock mLock;
 
   // Call worker thread. All access to mCall->Call() happens here.
   const RefPtr<AbstractThread> mCallThread;
